editable.sol

```diff
@@ -110,14 +112,42 @@ contract Editable is Ownable {
         emit UpdateRequest(objectHash);
     }
 
+    function removeVersionIdx(uint idx) {
+        delete versionHashes[idx];
+        delete versionTimestamp[idx];
+        if (idx != (versionHashes.length - 1)) {
+            versionHashes[idx] = versionHashes[versionHashes.length - 1];
+            versionTimestamp[idx] = versionTimestamp[versionTimestamp.length - 1];
+        }
+        versionHashes.length--;
+        versionTimestamp.length--;
+        return;
+    }
+
     function deleteVersion(string _versionHash) public returns (int256) {
         require(canCommit());
 
         bytes32 findHash = keccak256(abi.encodePacked(_versionHash));
         bytes32 objHash = keccak256(abi.encodePacked(objectHash));
         if (findHash == objHash) {
-            objectHash = "";
-            objectTimestamp = 0;
+            if (versionHashes.length == 0) {
+                objectHash = "";
+                objectTimestamp = 0;
+            } else {
+                // find the most recent
+                uint256 mostRecent = 0;
+                uint latestStamp = 0;
+                for (uint256 x = 0; x < versionHashes.length; x++) {
+                    if (versionTimestamp[x] > latestStamp) {
+                        mostRecent = x;
+                        latestStamp = versionTimestamp[x];
+                    }
+                }
+                // assign most recent version as object version and delete from versions array
+                objectHash = versionHashes[mostRecent];
+                objectTimestamp = latestStamp;
+                removeVersionIdx(mostRecent);
+            }
             emit VersionDelete(contentSpace, _versionHash, 0);
```

Comment:
should the event call ` emit VersionDelete(contentSpace, _versionHash, 0);` be added to `function removeVersionIdx(uint idx){}` to display right index?

Reply:
I think what's there is correct right now. When the hash to be removed is the 'current' value then it's logically ordinal zero.

Actually, with the way the implementation has changed, emitting the index doesn't have that much meaning anymore. Perhaps we should emit the timestamp.

Reply:
Got it. Also the function is called at line 159, should we emit the event at that line as well.